### ISSUE TITLE
BrainVISA may crash in casa_distro because of OpenGL

### ISSUE THREAD

**Issue by sapetnioc** [brainvisa/casa-distro#65]
Runninng `casa_distro run brainvisa` in 3.0 branch (may be the same in 2.*) via x2go crash with the following error :
```[9850:9915:0602/101950.753023:ERROR:gl_surface_qt.cpp(241)] GLX 1.3 or later is required.
[9850:9915:0602/101950.754317:ERROR:gl_surface_qt.cpp(371)] GLContextHelper::getEGLDisplay() failed.
WARNING: Application calling GLX 1.3 function "glXCreatePbuffer" when GLX 1.3 is not supported!  This is an application bug!
QSGContext::initialize: stencil buffer support missing, expect rendering errors
[9850:9915:0602/101952.122072:ERROR:gl_surface_qt.cpp(667)] Requested OpenGL implementation is not supported. Implementation: 0
*** stack smashing detected ***: <unknown> terminated
Aborted (core dumped)
```
This is an brain new opensource build workflow using Python 3 (same problem with Python 2) and bug_fix branch.
Not that it works via `ssh -X` and Anatomist works too (just tried to open an empty window).

I put the issue here because I believe this is not a BrainVISA related issue but is about how to make casa_distro useable in any execution context.

**Comment by ylep** [comment]
Did you test if you can run a simple OpenGL app like `glxgears`, both on the host system through `x2go`, and in the singularity container?

Also, just to make sure, were you using the currently distributed `cati/casa-dev:ubuntu-18.04` image from the `master` branch, or the experimental `casa-dev:ubuntu-18.04` image from the `runtime_image` branch?

As a side note, I do not know which approch `x2go` uses for remote OpenGL programs, but for running OpenGL programs remotely with hardware acceleration, I usually use `turbovnc` along with `virtualgl`. That approach requires to run OpenGL programs through a `vglrun` wrapper that uses tricks like `LD_PRELOAD` to force its own implementation of OpenGL. Maybe a similar trick is needed under `x2go`?

**Comment by sapetnioc** [comment]
glxgears works fine with or without casa_distro.

I work on 3.0 specific images that are built with the same shell scripts as the runtime_image branch but are not using Docker. I did not copy yet the OpenGL libraries because the `build_image.py` that replaces `Dockerfile` were first done for VirtualBox only. This may be the cause of the problem. I will put these libraries back in the v3.0 images later.

I tried several VNC solutions : TigerVNC, TightVNC and X2Go (but not TurboVNC nor virtualgl). I chose X2Go because it was the most stable solution to use the VirtualBox GUI (although it still crashes from time to time).
